### PR TITLE
feat: make subject and subtitle overridable

### PR DESCRIPTION
**Pull request by wilau2**
~Attempt 1~

~Moving title and subtitle outside of the Email editor. The reasoning is quite simple; those are the only fields not really part of the MJML/HTML. The fields are usually required from the APIs like Mailgun to send the email. Also, if you build a system where an email template is linked to an SMS template or WhatsApp template, the field would probably live in another place from a UI/UX perspective.~

~Right now, I did the "simple fix", but we could eventually delete those fields from the handled email template form.~

image

~Attempt 2~

~After feedback we are using the `BlockAttributeConfigurationManager` to override the subject and subtitle blocks.~
```
BlockAttributeConfigurationManager.add({
  [BasicType.SUBJECT]: () => <></>,
  [BasicType.SUBTITLE]: () => <></>,
})
```
image

**Attempt 3**

Subject and subtitle were not considered blocks, so we changed for boolean attributes on the component instead

```
BlockAttributeConfigurationManager.add({
  [BasicType.PAGE]: () => <DefaultPageConfigPanel hideSubTitle hideSubject />,
})
```

### CODE REVIEW

**Comment by m-Ryan** [comment]
image

The current recommended practice is that users can override the original configuration. Maybe we should provide an easier way instead of breaking change ?

**Comment by wilau2** [comment]
I'll explore other solutions. 
Maybe passing a configuration an optional configuration option to the editor standard layout component 🤔

**Comment by wilau2** [comment]
@m-Ryan Thanks for the feedback, I can now remove those with the override you suggested. 

```
BlockAttributeConfigurationManager.add({
  [BasicType.SUBJECT]: () => <></>,
  [BasicType.SUBTITLE]: () => <></>,
})
```

**Comment by m-Ryan** [comment]
Maybe we can do this by adding two parameters only to the Page component. What do you think? @wilau2 


```tsx
const DefaultPageConfigPanel  = BlockAttributeConfigurationManager.get(BasicType.PAGE)
BlockAttributeConfigurationManager({
   [BasicType.PAGE]: () => <DefaultPageConfigPanel hideSubject hideSubTitle />
})

```

We only need to update Page component.

const {hideSubTitle, hideSubject} = props

```tsx
hideSubTitle && {
             <TextField
                label={t('SubTitle')}
                name={'subTitle'}
                inline
              />
}
```

**Comment by wilau2** [comment]
The hide param would do the job in my case but I thought that using the block override would make it more flexible for other use cases.

**Comment by m-Ryan** [comment]
> The hide param would do the job in my case but I thought that using the block override would make it more flexible for other use cases.

🤔 But subject and subTitle are not a block, it looks a bit strange

**Comment by wilau2** [comment]
Let me try a third attempt at this :P

File: packages/easy-email-extensions/src/AttributePanel/utils/BlockAttributeConfigurationManager.ts [hunk 1]
```diff
@@ -1,6 +1,6 @@
 import { blocks } from '../components/blocks';
 
-type ObjectComponent = { [key: string]: () => JSX.Element | null };
+type ObjectComponent = { [key: string]: (props: any) => JSX.Element | null };
```
Comment (wilau2) [inline packages/easy-email-extensions/src/AttributePanel/utils/BlockAttributeConfigurationManager.ts:3]: Page is not the first component to have props. I'll check for better typing here.

**Comment by wilau2** [comment]
Hey @m-Ryan , I would appreciate if you could publish a release. 

Thanks a lot !

**Comment by m-Ryan** [comment]
image

The release has been published, and there are many file changes in this version. Please be sure to check them.

**Pull request merged**